osquery/tables/system/windows/services.cpp

```diff
@@ -0,0 +1,144 @@
+
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+#include <Winsvc.h>
+#include <string>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#pragma comment(lib, "Advapi32.lib")
+
+namespace osquery {
+namespace tables {
+
+const std::string kSvcStartType[] = {
+    "BOOT_START", "SYSTEM_START", "AUTO_START", "DEMAND_START", "DISABLED"};
+
+const std::string kSvcStatus[] = {"UNKNOWN",
+                                  "STOPPED",
+                                  "START_PENDING",
+                                  "STOP_PENDING",
+                                  "RUNNING",
+                                  "CONTINUE_PENDING",
+                                  "PAUSE_PENDING",
+                                  "PAUSED"};
+
+const std::map<int, std::string> kServiceType = {
+    {0x00000010, "OWN_PROCESS"},
+    {0x00000020, "SHARE_PROCESS"},
+    {0x00000100, "INTERACTIVE_PROCESS"},
+    {0x00000110, "OWN_PROCESS   (Interactive)"},
+    {0x00000120, "SHARE_PROCESS (Interactive)"}};
+
+SC_HANDLE schSCManager;
+
+BOOL QuerySvcInfo(ENUM_SERVICE_STATUS_PROCESS& svc, Row& r) {
+  SC_HANDLE schService;
+  LPQUERY_SERVICE_CONFIG lpsc = nullptr;
+  LPSERVICE_DESCRIPTION lpsd = nullptr;
+  DWORD cbBufSize = 0;
+
+  schService =
+      OpenService(schSCManager, svc.lpServiceName, SERVICE_QUERY_CONFIG);
+
+  if (schService == NULL) {
+    TLOG << "OpenService failed (" << GetLastError() << ")";
+    return FALSE;
+  }
+
+  (void)QueryServiceConfig(schService, NULL, 0, &cbBufSize);
+  lpsc = (LPQUERY_SERVICE_CONFIG)LocalAlloc(LMEM_FIXED, cbBufSize);
```

Comment:
Difference between Windows API memory allocation functions:

https://msdn.microsoft.com/en-us/library/windows/desktop/aa366533(v=vs.85).aspx

Essentially, `LocalAlloc` and `GlobalAlloc` are residual functions, from back in the day of 16-bit computing (Here's some more and better detailed history: http://stackoverflow.com/a/34327057), where one needed to differentiate between far and near memory pointers to allocated heap blocks..  Both of these are now just wrappers on top of `HeapAlloc`, however pay attention in the first article to this point:

> For example, HeapAlloc can be instructed to raise an exception if memory could not be allocated, a capability not available with LocalAlloc. LocalAlloc supports allocation of handles which permit the underlying memory to be moved by a reallocation without changing the handle value, a capability not available with HeapAlloc.

So with that information, let's dig into why would one use `malloc` vs `HeapAlloc`.  This largely boils down to when would one desire using a Windows API function call vs a native memory allocation. In larger Windows Applications it can be occasionally advantageous to allocate memory in one module or library, pass this on to another library or module, and have the latter module be held responsible for the deallocation. `HeapAlloc` gives this functionality where as with `malloc` you can hit potential memory corruption. Here's some much better versed explanation: http://stackoverflow.com/a/8224387

Since we're only allocating memory in this table and then quickly deallocating, `malloc` should be fine here.

Hope that helps!